sdc/datatypes/hpat_pandas_dataframe_functions.py

```diff
@@ -1627,3 +1628,108 @@ def sdc_pandas_dataframe_groupby_impl(self, by=None, axis=0, level=None, as_inde
         return init_dataframe_groupby(self, column_id, grouped, sort)
 
     return sdc_pandas_dataframe_groupby_impl
+
+
+def df_add_column_codelines(self, key):
+    """Generate code lines for DF add_column"""
+    func_lines = []
+    if self.columns:
+        func_lines += df_length_codelines(self)
+        func_lines += [
+            '  if length == 0:',
+            '    raise SDCLimitation("Could not set item for DataFrame with empty columns")',
+            '  elif length != len(value):',
+            '    raise ValueError("Length of values does not match length of index")',
+        ]
+    else:
+        func_lines += ['  length = len(value)']
+    func_lines += df_index_codelines(self)
+
+    results = []
+    for i, col in enumerate(self.columns):
+        res_data = f'res_data_{i}'
+        func_lines += [
+            f'  data_{i} = get_dataframe_data(self, {i})',
+            f'  {res_data} = pandas.Series(data_{i}, index=res_index, name="{col}")',
```

Comment:
I'm lost, where you declare "res_index"?

Reply:
`res_index` is defined inside of `df_index_codelines()`. Maybe it make sense to use expression `self.index` instead of `res_index`. @akharche what do you think about that?

Reply:
Got it! But imho it is not clear. May be, df_index_codelines() should return the appropriate index codeline that is set to res_index

Reply:
You are right, but again, maybe it make sense to use expression `self.index` instead of `res_index`?

Reply:
I can't use `self.index`, because in case of empty input DataFrame I should define length of index based on length of input value used for setting column. I left comment in the code about what is returned by `df_index_codelines()`.